core/src/setup/Kokkos_Setup_HIP.hpp

```diff
@@ -43,6 +43,11 @@
 #endif
 // clang-format on
 
+// The implementation of hipGraph in ROCm 5.2 is bugged, so we cannot use it.
+#if !((HIP_VERSION_MAJOR == 5) && (HIP_VERSION_MINOR == 2))
+#define KOKKOS_HIP_NATIVE_GRAPH 1
```

Comment:
@Rombur  ?

Reply:
> This naming (macro KOKKOS_BACKEND_BLA) is a bit unorthodox.
We typically say "enabled" followed by underscore backend and we don't actually give it a value.

Usually we use "enabled" when the user has control over the macro through CMake. Here there is nothing that the user can do, we just check the ROCm version. That's why we went with `KOKKOS_HIP_NATIVE_GRAPH`. I am fine using "enabled" for consistency.

> Do we intend it to be public?

Yes, it needs to be public because it changes the behavior of public functions.

> If so, should we including it in the information outputted by HIP::print_configuration()?

I would say no. It's useless if someone reports a bug because we already know that information thanks to the ROCm version. The macro will always be defined except for ROCm  5.2 which is not available on Frontier anymore.

Reply:
This was tackled in #7378. Closing.